vortex-file/src/read/mask.rs

```diff
@@ -121,12 +121,17 @@ impl RowMask {
 
     /// Combine the RowMask with bitmask values resulting in new RowMask containing only values true in the bitmask
     pub fn and_bitmask(self, bitmask: Array) -> VortexResult<Self> {
-        // TODO(robert): Avoid densifying sparse values just to get true indices
-        let sparse_mask =
-            SparseArray::try_new(self.to_indices_array()?, bitmask, self.len(), false.into())?
-                .into_array()
-                .into_bool()?;
-        Self::from_mask_array(sparse_mask.as_ref(), self.begin(), self.end())
+        // If we are a dense all true bitmap just take the bitmask array
+        if bitmask.len() as u64 == self.values.cardinality() {
```

Comment:
I feel like we should assert somewhere that `self.len() == self.values.cardinality()`. In the else branch SparseArray::try_new will verify that indices and bit mask have the same length.